lxd/storage/utils.go

```diff
@@ -1054,20 +1093,12 @@ func VolumeUsedByProfileDevices(s *state.State, poolName string, projectName str
 		// Iterate through each of the profiles's devices, looking for disks in the same pool as volume.
 		// Then try and match the volume name against the profile device's "source" property.
 		for name, dev := range profile.Devices {
-			if dev["type"] != cluster.TypeDisk.String() {
-				continue
-			}
-
-			if dev["pool"] != poolName {
-				continue
-			}
-
-			_, _, _, volumeName, err := DiskVolumeSourceParse(dev["source"])
+			usesVol, err := volumeIsUsedByDevice(*vol, nil, dev)
```

Comment:
This probably should correctly pick up `virtual-machine`/`container` volumes when attached as the root device of a profile.

Reply:
@tomponline I will open a new PR for this as it will require another DB query... we need a list of the instances using each profile. I do think that from a correctness standpoint `VolumeUsedByProfileDevices` should call its callback when the volume is associated with a Profile's root disk device.

Reply:
> I will open a new PR for this as it will require another DB query... we need a list of the instances using each profile

Why do you think this is needed? Isn' the job of `VolumeUsedByProfileDevices` to match disk devices in profiles to an instance device, not other instances? That is what `VolumeUsedByInstanceDevices` is for right?

> I do think that from a correctness standpoint VolumeUsedByProfileDevices should call its callback when the volume is associated with a Profile's root disk device.

Yeah that makes sense, as the profile is using that disk.